src/hooks/useDragLeave.ts

```diff
@@ -0,0 +1,57 @@
+import { debounce } from 'lodash'
+import { useEffect, useRef } from 'react'
+import { useDispatch } from 'react-redux'
+import { updateHoveringPathActionCreator } from '../actions/updateHoveringPath'
+import useSortedContext from './useSortedContext'
+
+const DEBOUNCE_DELAY = 50
+let hoverCount = 0
```

Comment:
Wouldn't this hook fail to work properly if used in multiple components, since we're using module-scoped variables here? I believe `hoverCount` should be tracked within the scope of the function rather than up at the top here.

Reply:
@trevinhofmann  We might have to turn `hoverCount` in a ref or state, so the value isn't reset on every render since it is being updated inside a `useEffect` hook.

Reply:
@trevinhofmann should we move the `useDragLeave` hook code into the `Thought` component where its called? The hook likely will not be called into other components, please correct me if I am wrong though. Thanks

Reply:
I would prefer to leave `useDragLeave` as a separate file which and cleaning it up just a bit (e.g. move `hoverCount` and `debouncedSetHoveringPath` into the function, likely as states or refs) so that it _could_ be reused elsewhere. Adding large hooks to a React component's file adds too much complexity in one place for my preference.

@raineorshine -- Do you agree?